### DISCUSSION TITLE
How to analysze smoothness of a graph?

### DISCUSSION THREAD

**Discussion by Kevin-shihello-world** [pyg-team/pytorch_geometric#5570] [Ideas]
Nowadays more and more GNN researchers are paying attention to the over-smooth problem of GNNs. When we stack GNN layers higher those node features tend to get similar to each other and be hard to extract useable information from. So I want to do research on this and found there seems only exists one application of count smoothness for GNN in CS-GCN code, which seems only applies to graphs in datasets like cora, I mean nodes in those graphs have less feature dimension and code for them may be hard to apply to graphs with higher node-feature-dimension like progressed graphs in GNN layer. 
So do you have a better way to count the smoothness of graphs?(especially to count them in lower compute cost)
I thought if the over-smooth problem of GNNs means GNNs can hardly extract information from over-smoothed graphs then we may be able to count the smoothness by counting the difference of node features of a graph before and after an aggregate operation in PYG.messagepassing it would be efficient and needs lower compute cost. Am I right? Have I missed something?
 I would appreciate it if anyone can offer me some tips or inspiration for this question. Thank you!

**Comment by EdisonLeeeee** [comment]
You can take a look at this paper: https://arxiv.org/abs/1909.03211

**Reply by Kevin-shihello-world** [reply]
Thanks, I forgot I've seen that one before but its algorithm is too slow compared to my gcnconv one.
 I want to set this question to be an open question and mark every useful piece of advice as an answer.

**Reply by Kevin-shihello-world** [reply]
The algorithm in the paper you advised has to calculate pair-wise cos-distance and eliminate some of them using masks
